src/app.ts

```diff
@@ -1,24 +1,19 @@
-import { Image } from "./image/image.ts";
 import { AppComponent } from "./components/app-component.ts";
-import { AppState, Store } from "./component.ts";
+import { Controller } from "./controller.ts";
+import { AppController } from "./controllers/app-controller.ts";
+import { SpriteDetailsController } from "./controllers/sprite-details-controller.ts";
 
 const init = () => {
-  // handle all your on load setup here
-  const _newImage = new Image("<put path here>", "<put name here>");
-  // will be useful in the future
-  // window.alert(`
-  //   here is where you would implement image uploading,
-  //   which will give you a path,
-  //   which you use in the constructor of the image,
-  //   which will have the name ${newImage.name}
-  // `);
-};
+  Controller.registerController(
+    AppController.typeName,
+    new AppController(),
+  );
+  Controller.registerController(
+    SpriteDetailsController.typeName,
+    new SpriteDetailsController(),
+  );
 
-const store = new Store(new AppState(), new AppComponent());
-store.update();
+  new AppComponent().render();
+};
 
-if (document.readyState !== "loading") {
-  init();
-} else {
-  self.addEventListener("DOMContentLoaded", init);
-}
+init();
```

Comment:
I changed this because this was very much not working with Deno native apps... But it works for browsers, so I can revert this and move the `init()` back inside the events if we prefer.

Reply:
yeah you'll need to as it'll race, and if init is called before the events there will be no DOM to render the components to.